packages/ui/src/FileCard/index.tsx

```diff
@@ -0,0 +1,170 @@
+import { Button } from "primereact/button";
+import { Card } from "primereact/card";
+import { ReactNode } from "react";
+
+import { IFile } from "../FilesTable";
+
+export type Messages = {
+  downloadCountLabel?: string;
+  downloadButtonLabel?: string;
+  lastDownloadedAtHeader?: string;
+  uploadedbyHeader?: string;
+  uploadedAtHeader?: string;
+  viewButtonLabel?: string;
+};
+
+type FileCardType = {
+  file: IFile;
+  messages?: Messages;
+  onArchive?: (arguments_: IFile) => void;
+  onDelete?: (arguments_: IFile) => void;
+  onDownload?: (arguments_: IFile) => void;
+  onShare?: (arguments_: IFile) => void;
+  onView?: (arguments_: IFile) => void;
+  renderThumbnail?: (arguments_: IFile) => ReactNode;
+  showDescription?: boolean;
+  showEditDescription?: boolean;
+  showSize?: boolean;
+  showThumbnail?: boolean;
+};
+
+export const FileCard = ({
+  file,
+  onArchive,
+  onDelete,
+  onDownload,
+  onShare,
+  onView,
+  messages,
+  renderThumbnail,
+  showDescription = true,
+  showEditDescription = true,
+  showSize = true,
+  showThumbnail = true,
+}: FileCardType) => {
+  const actionButtons: any[] = [];
+
+  const renderThumbnailType = () => {
+    if (!showThumbnail) {
+      return null;
+    }
+    return (
+      <div className="file-thumbnail">
+        {renderThumbnail ? (
+          renderThumbnail(file)
+        ) : (
+          <i className="pi pi-file-pdf"></i>
+        )}
+      </div>
+    );
+  };
+
+  const checkUploadedByData = (data: any) => {
+    if (!data.uploadedBy) {
+      return <code>&#8212;</code>;
+    }
+
+    if (data.uploadedBy.givenName || data.uploadedBy.lastName) {
+      return `${data.uploadedBy.givenName || ""} ${
+        data.uploadedBy.lastName || ""
+      }`;
+    }
+
+    return data.uploadedBy.email;
+  };
+
+  const updateDateFormat = (date: number) => {
+    return new Date(date).toLocaleDateString("en-GB");
+  };
```

Comment:
Can we rename this to `formatDate` and move this to some global util file. Also the locale should be configurable i.e come in as argument. We should pass the current app's locale to this argument.